// bycf/app.mjs
const BASE_URL = 'https://shannz.zone.id';

async function callBypassAPI(endpoint, data = {}) {
    const validEndpoints = ['stats', 'waf-session', 'solve-turnstile-min', 'solve-turnstile-max', 'get-source'];
    if (!validEndpoints.includes(endpoint)) {
        throw new Error(`Endpoint tidak valid: ${endpoint}. Harap gunakan salah satu dari: ${validEndpoints.join(', ')}`);
    }

    const url = `${BASE_URL}/api/${endpoint}`;
    let options = {
        headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json'
        }
    };

    if (endpoint === 'stats') {
        options.method = 'GET';
    } else {
        options.method = 'POST';
        if (Object.keys(data).length > 0) {
            options.body = JSON.stringify(data);
        } else if (['waf-session', 'solve-turnstile-max', 'get-source'].includes(endpoint)) {
             if (!data.url) {
                throw new Error(`Parameter 'url' wajib untuk endpoint /api/${endpoint}`);
            }
        }
    }

    try {
        const response = await fetch(url, options); 
        const json = await response.json();
        
        if (json.success === true) {
            return json.data;
        } 

        const errorMessage = json.error || `Error API tidak diketahui pada endpoint /api/${endpoint}. Status: ${response.status}`;
        throw new Error(errorMessage);

    } catch (error) {
        console.error(`Gagal memanggil API ${endpoint}:`, error.message);
        throw error;
    }
}

const shannz = {
    stats: () => callBypassAPI('stats'),
    wafSession: (url, proxy) => callBypassAPI('waf-session', { url, proxy }),
    turnstileMin: (url, siteKey, proxy) => callBypassAPI('solve-turnstile-min', { url, siteKey, proxy }),
    turnstileMax: (url, proxy) => callBypassAPI('solve-turnstile-max', { url, proxy }),
    source: (url, proxy) => callBypassAPI('get-source', { url, proxy })
};

export { shannz };
